modules/host/negotiate.go

```diff
@@ -280,6 +280,10 @@ func (h *Host) rpcRevise(conn net.Conn) error {
 	if err := encoding.ReadObject(conn, &fcid, crypto.HashSize); err != nil {
 		return errors.New("couldn't read contract ID: " + err.Error())
 	}
+
+	// remove conn deadline while we wait for lock and rebuild the Merkle tree
+	conn.SetDeadline(time.Time{})
```

Comment:
fighting for the obligation lock could take a while, but in practice it shouldn't happen. I haven't run Merkle benchmarks in a while, but it could definitely take a bit for a large file.